pkg/runtime/wfengine/workflow.go

```diff
@@ -52,7 +39,29 @@ const (
 	AddWorkflowEventMethod       = "AddWorkflowEvent"
 )
 
-func NewWorkflowActor(scheduler workflowScheduler) actors.InternalActor {
+type workflowActor struct {
+	actors          actors.Actors
+	states          map[string]*workflowState
+	scheduler       workflowScheduler
+	rwLock          sync.RWMutex
+	cachingDisabled bool
+}
+
+type durableTimer struct {
+	Bytes      []byte
+	Generation uuid.UUID
+}
+
+func init() {
+	// gob is used to serialize internal actor reminder data
+	gob.Register(durableTimer{})
```

Comment:
Could you share more details about it? I do not get the benefit of it.

Reply:
Please see my PR description [here](https://github.com/dapr/dapr/pull/5344), where I talk about the reasons for introducing the gob dependency. If you're specifically asking about why I need to call `gob.Register(...)`, it's because reminder payloads are defined as `interface{}` (see [create_reminder_request.go](https://github.com/dapr/dapr/blob/a738f7f200036c200236fa83856ce5b8d8c79ed5/pkg/actors/create_reminder_request.go#L21)), so any structs that get serialized by `gob` have to be declared explicitly. See [here](https://stackoverflow.com/questions/32676898/whats-the-purpose-of-gob-register-method) for more info on this.